### PR TITLE
feat(module:timeline): support custom position

### LINKED ISSUE #5470: Timeline with custom alternate
## What problem does this feature solve?
Would be nice to have the option to define of which side of the timeline a timeline-item should appear. For example, during a football game a team scores twice. This means that two items should appear on the same side of the timeline.

## What does the proposed API look like?
nz-timeline should have a new `[nzMode]` called `'custom'`.
nz-timeline-item should have a new property `[nzSide]` with type `'left'` or `'right'`. If `[nzSide]` is not informed, it should appear on the same side as the previous timeline-item.

### PR DESCRIPTION
**Pull request by wendellhu95**
close #5470

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: components/timeline/doc/index.en-US.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ Timeline
 | `[nzPending]` | Set the last ghost node's existence or its content | `string\|boolean\|TemplateRef<void>` | `false` |
 | `[nzPendingDot]` | Set the dot of the last ghost node when pending is true | `string\|TemplateRef<void>` | `<i nz-icon nzType="loading"></i>` |
 | `[nzReverse]` | Reverse nodes or not | `boolean` | `false` |
-| `[nzMode]` | By sending `alternate` the timeline will distribute the nodes to the left and right | `'left' \| 'alternate' \| 'right'` | - |
+| `[nzMode]` | By sending `alternate` the timeline will distribute the nodes to the left and right | `'left' \| 'alternate' \| 'right' \| 'custom'` | - |
```
Comment (hsuanxyz) [inline components/timeline/doc/index.en-US.md:38]: React 只在 `alternate` 下生效。https://github.com/ant-design/ant-design/blob/feature/components/timeline/Timeline.tsx#L50。

我觉得没必要增加新模式，没有 `nzPosition` 的回退到 `nzMode` 的行为。

Comment (hsuanxyz) [reply components/timeline/doc/index.en-US.md:38]: 他们有关于 `mode` 的讨论，https://github.com/ant-design/ant-design/pull/16193

Comment (wendellhu95) [reply components/timeline/doc/index.en-US.md:38]: 感觉是为了图方便，因为需要 `ant-timeline-alternate` 的样式 🤣  语义上来说 alternate 是交替

Comment (hsuanxyz) [reply components/timeline/doc/index.en-US.md:38]: 🉑️

File: components/timeline/timeline.component.ts [hunk 2]
```diff
@@ -139,3 +133,18 @@ export class NzTimelineComponent implements AfterContentInit, OnChanges, OnDestr
...
+      ? undefined
+      : mode === 'left'
+      ? 'left'
+      : mode === 'right'
+      ? 'right'
+      : mode === 'alternate' && index % 2 === 0
+      ? 'left'
+      : 'right';
+
+  return position as NzTimelinePosition | undefined;
```
Comment (hsuanxyz) [inline components/timeline/timeline.component.ts:149]:
```suggestion
  return position;
```
看上去不是必须的

**Pull request merged**